File: module-2-contents/src/main/java/com/codurance/first/thorough/password/PasswordValidator.java
package com.codurance.first.thorough.password;

public class PasswordValidator {
    private final int MIN_LENGTH = 6;
    private final int MAX_LENGTH = 20;

    public boolean isValid(String password) {
        return password.length() >= MIN_LENGTH && password.length() <= MAX_LENGTH;
    }
}

File: password-validation/src/PasswordValidator/PasswordValidator.ts
export class PasswordValidator {
    public execute(password: string) {
        if (password.length <= 8) {
            return false;
        }

        if (!/[A-Z]/.test(password)) {
            return false;
        }

        if (!/\d/.test(password)) {
            return false;
        }

        if (!/_/.test(password)) {
            return false;
        }

        if (!/[a-z]/.test(password)) {
            return false;
        }

        return true;
    }
}